src/tools/compiletest/src/runtest.rs

```diff
@@ -1420,6 +1457,11 @@ impl<'test> TestCx<'test> {
             rustc.arg(dir_opt);
         };
 
+        // `use-minicore` requires `#![no_std]` and `#![no_core]`, which means no unwinding panics.
+        if self.props.use_minicore {
```

Comment:
Remark: technically, this also needs to check and ensure that the test writer does not manually override `-C panic=xxx` where `xxx != abort`, but that should happen after compiletest directive handling rework.